mne/io/tests/test_meas_info.py

```diff
@@ -456,4 +460,33 @@ def test_csr_csc():
     assert_array_equal(ct_read.toarray(), ct.toarray())
 
 
+@testing.requires_testing_data
+def test_check_compensation_consistency():
+    """Test check picks compensation."""
+    raw = read_raw_ctf(ctf_fname, preload=False)
+    events = make_fixed_length_events(raw, 99999)
+    picks = pick_types(raw.info, meg=True, exclude=[], ref_meg=True)
+    pick_ch_names = [raw.info['ch_names'][idx] for idx in picks]
+    for (comp, expected_result) in zip([0, 1], [False, False]):
+        raw.apply_gradient_compensation(comp)
+        ret, missing = _bad_chans_comp(raw.info, pick_ch_names)
+        assert ret == expected_result
+        assert len(missing) == 0
+        Epochs(raw, events, None, -0.2, 0.2, preload=False, picks=picks)
+
+    picks = pick_types(raw.info, meg=True, exclude=[], ref_meg=False)
+    pick_ch_names = [raw.info['ch_names'][idx] for idx in picks]
+
+    for (comp, expected_result) in zip([0, 1], [False, True]):
+        raw.apply_gradient_compensation(comp)
+        ret, missing = _bad_chans_comp(raw.info, pick_ch_names)
+        assert ret == expected_result
+        assert len(missing) == 17
+        if comp != 0:
+            pytest.raises(RuntimeError, Epochs, raw, events, None, -0.2, 0.2,
```

Comment:
FYI the "modern" way to do this is:
```
with pytest.raises(RuntimeError, match='something from the error string'):
    Epochs(...)
```
It is a bit more precise than using `pytest.raises` as a drop-in for `assert_raises`